src/Game.java

```diff
@@ -1,30 +1,416 @@
+            for(Card.Colour c:Card.Colour.values()){
+                if (c.toString().equals(Card.Colour.WILD.toString())){
+                    continue;
+                }
+                if(input.toString().equals(c.toString())){
+                    System.out.println("The colour is now "+c);
+                    return c;
+                }
+            }
+            if(!colourSet) {
+                System.out.println("Invalid colour");
+            }
+
+        }
+        return null;
+
+
+
+    }
+
+    /**
+     * Check if the player has no cards remaining
+     *
+     * @param player The player that is checked
+     * @return True if the player has no cards, false otherwise
+     */
+    private boolean isWinner(Player player){
+        return player.getHandSize()==0;
+    }
+
+    /**
+     * Go to the turn of the next player based on turn direction
+     */
+    public void nextTurn(){
+        //clockwise
+        if(turnDirection){ //0->1
+            currentTurn = (currentTurn+1) % numPlayers;
+            nextPlayerIndex = (currentTurn+1) % numPlayers;
+        }
+        //counterclockwise
+        else{ //0->3->2->1
+            currentTurn = (currentTurn-1 + numPlayers)%numPlayers;
+            nextPlayerIndex = (currentTurn-1+numPlayers)%numPlayers;
+
+            }
+
+
+        }
+
+
+    /**
+     * Skip the turn of the next player
+     */
+    public void skipTurn(){
+        nextTurn();
+        nextTurn();
+    }
+
+    /**
+     * Draw the amount of cards(n) based on which player(index) will be receiving them
+     * @param n The amount of cards to be added to the hand of the player
+     * @param index The index of the player that will be receiving cards
+     */
+    public void drawNCards(int n,int index){
+        players.get(index).addCardToHand(n);
+        if(index==currentTurn){
+            System.out.println(players.get(index).getName()+" has drawn a card:"+players.get(index).getCard(players.get(index).getHandSize()-1));
+        }
+        else{
+            System.out.println(players.get(index).getName()+" has to draw "+n+" card(s) due to "+topCard.toString());
+        }
+    }
+
+
+    /**
+     * Set the current colour of the game
+     * @param colour The colour to be set as the current colour
+     */
+    public void setCurrentColour(Card.Colour colour){
+        currentColour = colour;
+
+    }
+
+    /**
+     * Returns the score of the winner
+     *
+     * @return The score of the winner
+     */
+    private int getWinnerScore(){
+        int winnerScore = 0;
+        if(topCard.getRank().ordinal() == 8){
+            drawNCards(1,nextPlayerIndex);
+
+        }else if(topCard.getRank().ordinal() == 13){
+            //wild draw two is on top
+            drawNCards(2,nextPlayerIndex);
+        }
+
+        for(Player p: players){
+            winnerScore += p.getHandScore();
+        }
+        return winnerScore;
+    }
+
+    public static void main(String[] args) {
```

Comment:
Add a JavaDoc comment for the main method. 
/**
 * Our main method. Some kind of handy description goes here.
 * @param args The command line arguments.
 */